src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -35,13 +35,46 @@ function useDecline () {
   })
 }
 
+function useFetchCounterparties () {
+  const { data: { holofuelActionableTransactions = [] } = {} } = useQuery(HolofuelActionableTransactionsQuery)
+
+  const { loading, error, data: { holofuelInboxCounterparties } = {}, client } = useQuery(HolofuelInboxCounterpartiesQuery)
+
+  if (holofuelInboxCounterparties) {
+    const filterTransactionsByAgentId = (agent, txListType) => txListType.filter(transaction => transaction.counterparty.id === agent.id)
+    const updateTxListCounterparties = (txListType, counterpartyList) => counterpartyList.map(agent => {
+      const matchingTx = filterTransactionsByAgentId(agent, txListType)
+      return matchingTx.map(transaction => { Object.assign(transaction.counterparty, agent); return transaction })
+    })
+
+    const result = flatten(updateTxListCounterparties(holofuelActionableTransactions, holofuelInboxCounterparties))
+
+    client.writeQuery({
+      query: HolofuelActionableTransactionsQuery,
+      data: {
+        holofuelActionableTransactions: result
+      }
+    })
+  }
+
+  let response
+  if (loading) response = { loading: true }
+  else if (error) response = { error: `Error: ${error}` }
+  else response = holofuelInboxCounterparties
+  return response
+}
+
 export default function Inbox () {
   const { data: { holofuelActionableTransactions: transactions = [] } = {} } = useQuery(HolofuelActionableTransactionsQuery)
+
   const payTransaction = useOffer()
   const declineTransaction = useDecline()
   const [modalTransaction, setModalTransaction] = useState()
   const isTransactionsEmpty = isEmpty(transactions)
 
+  // const counterpartyList = useFetchCounterparties()
+  useFetchCounterparties()
```

Comment:
nitpicking, but I'd move this to line 69 so that it's next to the code it relates to